### PR TITLE
feat: create a new deployment script

### PR DESCRIPTION
**Pull request by barrutko**

### CODE REVIEW

File: scripts/deploy.ts [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+type CronTrigger = {
+    type: 'cron'
+    cron: string
+}
+
+type EventTrigger = {
+    type: 'event'
+    filter: {
+        address: string
+        topics: Array<{
```
Comment (lucas-manuel) [inline scripts/deploy.ts]: rename to something like eventInfo? Cause its used to parse info into `topics`

File: scripts/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    return Wallet.fromEncryptedJsonSync(keystore, password).privateKey
+}
+
+const getBufferFromString = (str: string): Buffer => {
+    return Buffer.from(str, 'utf-8')
+}
+
+const createHash = (rawConfig: Buffer, ipfsDeployment: string): string => {
+    const configHash = utils.keccak256(rawConfig)
+    const ipfsDeploymentHash = utils.keccak256(getBufferFromString(ipfsDeployment))
```
Comment (lucas-manuel) [inline scripts/deploy.ts]: Don't need to hash the ipfs ID, not keccak but still a hash

File: scripts/deploy.ts [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    return utils.keccak256(getBufferFromString(configHash.concat(ipfsDeploymentHash)))
+}
+
+const pause = () => {
+    return new Promise<void>((resolve) => {
+        prompter.question('Press ENTER to confirm & continue...', () => {
```
Comment (lucas-manuel) [inline scripts/deploy.ts:121]: Need to be able to disable in CI (add flag maybe)

File: scripts/deploy.ts [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+                continue
+            }
+            if (!triggerTypeNames.includes(config.trigger.type)) {
+                console.log(`Trigger type ${config.trigger.type} is not supported`)
+                continue
+            }
+
+            const configName = configFileName.slice(0, -5) // remove '.json'
+            const hash = createHash(rawConfig, ipfsDeployments[w3fName])
+            const deploymentName = `${configName} ${hash}`
```
Comment (lucas-manuel) [inline scripts/deploy.ts]: Underscore or dash

File: scripts/deploy.ts [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+            const secrets: Secrets = Object.entries(config.secrets).reduce((secrets, [key, envVarName]) => {
+                const value = process.env[envVarName]
+                if (!value) throw new Error(`${envVarName} env variable is not defined`)
+                secrets[key] = value
+                return secrets
+            }, {} as Record<string, string>)
+
+            console.log(`${addDryRunNotice(scriptArguments.dryRun)}Deploying ${deploymentName}`)
+            if (!scriptArguments.dryRun) {
+                await pause()
```
Comment (lucas-manuel) [inline scripts/deploy.ts]: Add flag

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: scripts/deploy.ts (hunk 1 above)
Comment (barrutko) [reply scripts/deploy.ts]: done

File: scripts/deploy.ts (hunk 2 above)
Comment (barrutko) [reply scripts/deploy.ts]: done

File: scripts/deploy.ts (hunk 3 above)
Comment (barrutko) [reply scripts/deploy.ts:121]: done

File: scripts/deploy.ts (hunk 4 above)
Comment (barrutko) [reply scripts/deploy.ts]: done

File: scripts/deploy.ts (hunk 5 above)
Comment (barrutko) [reply scripts/deploy.ts]: done

**Review by lucas-manuel — DISMISSED** [review]
Approving for now, will be ready for merge after running a full deploy with real keys once @hexonaut configures

**Pull request merged**